active/RowColumnAPIChange/rowcolumnapichange.md

```diff
@@ -0,0 +1,168 @@
+
+These overloaded constructors take the same arguments as the [GridLength](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.GridLength) constructor does - a double and a GridUnitType. In turn, feeding these same arguments to a RowDefinition or ColumnDefinition constructor will create a GridLength object and assign it to the ColumnDefinition's Width property or the RowDefinition's Height property behind the scenes. 
+
+# Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+Current syntax to define a Grid with one column in code-behind:
+```csharp
+var grid = new Grid();
+var column = new ColumnDefinition();
+column.Width = new GridLength(1.0, GridUnitType.Star);
+grid.ColumnDefinitions.Add(column);
+```
+
+or
+
+```csharp
+var column = new ColumnDefinition
+{
+    Width = new GridLength(1.0, GridUnitType.Star)
+};
+```
+
+
+New syntax to define a Grid with one column in code-behind:
+```csharp
+var grid = new Grid();
+grid.ColumnDefinitions.Add(new ColumnDefinition(1.0, GridUnitType.Star)); // Width="1*"
+```
+
+New syntax to define a Grid with one simple column in code-behind, where the GridUnitType is a pixel:
+```csharp
+grid.ColumnDefinitions.Add(new ColumnDefinition(500.0)); // Width="500px"
+```
+
+<!-- # Remarks -->
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+# API Notes
+ColumnDefinition constructor will be overloaded to take a double `pixelWidth` OR take two arguments consisting of a double `width` and a GridUnitType `type`:
+
+```csharp
+public ColumnDefinition(double pixelWidth) { ... }
+```
+
+```csharp
+public ColumnDefinition(double width, GridUnitType type) { ... }
+```
+
+RowDefinition constructor will be overloaded to take a double `pixelHeight` OR take two arguments consisting of a double `height` and a GridUnitType `type`:
+
+```csharp
+public RowDefinition(double pixelHeight) { ... }
+```
+
+```csharp
+public RowDefinition(double height, GridUnitType type) { ... }
```

Comment:
Why is one named pixelHeight and the other named height? I think we should be consistent here.

Reply:
`pixelHeight` makes sense to me. It makes it clear that when there's no type explicitly specified, pixels will be used, but when there is a type, you don't know the unit. The way I see it, intellisense should guide you, and a good parameter name makes it very clear what it does.

Reply:
Right, haven't thought of that. Having it just called height makes more sense here then.

Reply:
Yep, I believe that was the reasoning for calling it pixelHeight. 👍